storage/internal/benchmarks/client_pool.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/option"
+	htransport "google.golang.org/api/transport/http"
+	"google.golang.org/grpc"
+)
+
+type clientPool struct {
```

Comment:
Can we add some overall docs for this file? In particular would be good to clarify the purpose of having client pools, the number of clients we'd expect to be spun up, and the relationship between this and `-workers`.

Reply:
I added some - lmk if you see any gaps or think we should change how it currently works

Reply:
Discussed offline -- we probably want to have a more deterministic number of clients spun up to reduce variability between runs, rather than using a non-fixed size pool. @BrennaEpp will file an issue to make this change, but it's not a blocker for this PR.